### PR TITLE
[receiver/googlepubsub] fix span and trace id parsing crash

### PR DESCRIPTION
**Pull request by alexvanboxel**
**Description:** <Describe what has changed.>
Fixed a crash in the span and trace ID parsing when encountering corrupt data. Enhanced the `cloud_logging` parsing so it's better testable and includes additional tests. Made the code read for options in the parsing json/protoPayload (implementation not included).

Includes minor fixes in the README.

**Link to tracking Issue:**
#32007

**Testing:** 
Added additional tests for the bug (trace and span id), but also for several of the other mappings.

**Documentation:**
The documentation now also includes all of the mappings the receiver does in `cloud_logging` mode.

### CODE REVIEW

File: receiver/googlecloudpubsubreceiver/README.md [hunk 1]
```diff
@@ -89,3 +89,56 @@ AND
 attributes.content-type = "application/protobuf"
 ```
 
+# Google Cloud Logging
+
+**Status**: Experimental
+
+Google Cloud logging uses the [LogEntry] to cary log information. In this section the mapping of the fields to
```
Comment (tiffany76) [inline receiver/googlecloudpubsubreceiver/README.md:96]:
```suggestion
Google Cloud logging uses the [LogEntry] to carry log information. In this section, the mapping of the fields to 
```

File: receiver/googlecloudpubsubreceiver/README.md [hunk 2]
```diff
@@ -89,3 +89,56 @@ AND
...
+# Google Cloud Logging
+
+**Status**: Experimental
+
+Google Cloud logging uses the [LogEntry] to cary log information. In this section the mapping of the fields to 
+OpenTelemetry fields and attributes are documented.
+
+## Semantic Mapping
+
+Some of the attributes can be moved to OpenTelemetry Semantic Conventions. Note however that all the attributes are
```
Comment (tiffany76) [inline receiver/googlecloudpubsubreceiver/README.md:101]:
```suggestion
Some of the attributes can be moved to OpenTelemetry Semantic Conventions. Note, however, that all the attributes are
```

File: receiver/googlecloudpubsubreceiver/README.md [hunk 3]
```diff
@@ -89,3 +89,56 @@ AND
...
+considered experimental and are subject to change.
+
+| Field       | Type      | Description                            | Maps to Unified Model Field    |
+|-------------|-----------|----------------------------------------|--------------------------------|
+| `insert_id` | `boolean` | A unique identifier for the log entry. | `Attributes["log.record.uid"]` |
+
+## Attribute and Field mapping
+
+The rest of the JSON body is either mapped to attributes with an `gcp` prefix or directly mapped to the LogRecord.
```
Comment (tiffany76) [inline receiver/googlecloudpubsubreceiver/README.md:110]:
```suggestion
The rest of the JSON body is either mapped to attributes with a `gcp` prefix or directly mapped to the LogRecord. 
```

File: receiver/googlecloudpubsubreceiver/README.md [hunk 4]
```diff
@@ -89,3 +89,56 @@ AND
...
+| `httpRequest`      | `HttpRequest`            | The HTTP request associated with the log entry, if any.                                                          | `Attributes["gcp.http_request"]` (KVList)     |
+| `operation`        | `LogEntryOperation`      | Information about a operation associated with the log entry.                                                     | `Attributes["gcp.operation"]`  (KVList)       |
+| `sourceLocation`   | `LogEntrySourceLocation` | Source code location information associated with the log entry.                                                  | `Attributes["gcp.source_location"]` (KVList)  |
+| `split`            | `LogSplit`               | Information indicating this LogEntry is part of a sequence of multiple log entries split from a single LogEntry. | `Attributes["gcp.log_split"]`  (KVList)       |
+
+## Severity Mapping
+
+The severity is mapping from [Google Cloud Log Severity](https://cloud.google.com/logging/docs/reference/v2/rest/v2/LogEntry#LogSeverity) to the OpenTelemetry Severity Number
```
Comment (tiffany76) [inline receiver/googlecloudpubsubreceiver/README.md:132]:
```suggestion
The severity is mapping from [Google Cloud Log Severity](https://cloud.google.com/logging/docs/reference/v2/rest/v2/LogEntry#LogSeverity) to the OpenTelemetry Severity Number.
```

**Review by tiffany76 — COMMENTED** [review]
Tech writer here. Just a few tiny docs fixes. Thanks!

**Comment by alexvanboxel** [comment]
Rebase against, main (but... it's impossible to keep up). For the approvers : I am the CODEOWNER of this receiver

**Comment by Manuelraa** [comment]
Could we please reopen this in order to move forward with closing #33400 ?

**Pull request closed without merging**